app/routes/api/auth.ts

```diff
@@ -2,35 +2,28 @@ import type { RequestHandler } from 'express';
 import routerFactory from 'express-promise-router';
 const router = routerFactory();
 import passport from 'passport';
-import { handleError } from '../errors';
-import Register from '../models/register';
-
-router.get('/login', (req, res) => {
-  const csrfToken = process.env.NODE_ENV !== 'test' ? req.csrfToken() : 'test';
-  res.render('login', {
-    csrfToken: csrfToken,
-    feedback: req.flash('error'),
-  });
-});
+import { handleError } from '../../errors';
+import Register from '../../models/register';
 
 const stripUsername: RequestHandler = (req, res, next) => {
   req.body.username = req.body.username.trim();
   next();
 };
 
+router.get('/token', (req, res) => {
+  const csrfToken = req.csrfToken();
+  res.json({ csrfToken });
+});
+
```

Comment:
Could you explain this?

Reply:
The [synchronizer token pattern](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#synchronizer-token-pattern) sends a token from the server to the client as a payload. Calling the function generates that token and stores a token for the given session.

The client has to use the token on each subsequent request, ideally as a [custom header](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#synchronizer-token-pattern)

It works as intended on the signin-page, but i may have modify how it is fetched/stored client-side when we move more endpoints.

Reply:
Okey I see thanks for a super explanation! So the `req.csrfToken()` generates the token? I would have thought it retrieves it from the request based on the name 🙃